src/cursor/cur_backup.c

```diff
@@ -359,18 +359,18 @@ __backup_start(WT_SESSION_IMPL *session,
 	}
 
 err:	/* Close the hot backup file. */
-	if (cb->bfs != NULL)
-		WT_TRET(__wt_fclose(session, &cb->bfs));
 	if (srcfs != NULL)
 		WT_TRET(__wt_fclose(session, &srcfs));
 	if (ret == 0) {
 		WT_ASSERT(session, dest != NULL);
-		WT_TRET(__wt_fs_rename(session, WT_BACKUP_TMP, dest, false));
+		WT_TRET(__wt_sync_and_rename(session,
+		    &cb->bfs, WT_BACKUP_TMP, dest));
 		__wt_writelock(session, &conn->hot_backup_lock);
 		conn->hot_backup_list = cb->list;
 		__wt_writeunlock(session, &conn->hot_backup_lock);
 		F_SET(session, WT_SESSION_BACKUP_CURSOR);
-	}
+	} else if (cb->bfs != NULL)
+		WT_TRET(__wt_fclose(session, &cb->bfs));
```

Comment:
I'm also wondering about the fact that we ignore failures from `__wt_sync_and_rename()` for the purpose of setting `WT_CONNECTION.hot_backup_list`? I mean, why isn't it something like this:
```
err:    /* Close the hot backup file. */
        if (srcfs != NULL)
                WT_TRET(__wt_fclose(session, &srcfs));
        if (ret == 0)  
                ret = __wt_sync_and_rename(session,
                    &cb->bfs, WT_BACKUP_TMP, dest));
        if (ret == 0) {
                __wt_writelock(session, &conn->hot_backup_lock);
                conn->hot_backup_list = cb->list;
                __wt_writeunlock(session, &conn->hot_backup_lock);
                F_SET(session, WT_SESSION_BACKUP_CURSOR);
        }
        if (cb->bfs != NULL)
                WT_TRET(__wt_fclose(session, &cb->bfs));
```
I don't know how this code works, so if you're happy with it, fbm.

(I tossed `WT_ASSERT(session, dest != NULL);`. Not sure why it's there historically, but it can't fire in the current code.)